Add grammar for CartoCSS

floscher opened the pull request:
This pull request has to wait, until yohanboniface/carto-atom#1 is resolved. But I'm optimistic, that the owner will choose a suitable license. He [has noted in the package.json](https://github.com/yohanboniface/carto-atom/blob/master/package.json#L5) that he intended to release it under WTFPL.

In order to get this work, I had to change the `download-grammars`-script to accept files that end with `.json` instead of `.cson`.

arfon commented:
@aroben is [this](https://github.com/yohanboniface/carto-atom/blob/master/package.json#L5) good enough for our purposes?

arfon commented:
OK we're not waiting on the license any longer but I'd like @aroben to :+1: the changes to the download script.

aroben commented on script/download-grammars:
```diff
@@ -139,7 +139,7 @@ def load_grammar(path)
     Plist::parse_xml(path)
-  when '.cson'
+  when '.cson', '.json'
     cson = `"#{CSONC}" "#{path}"`
```
There's no need to pass `.json` files through `csonc`. It would be slightly nicer to just go directly to `JSON.parse` in this case.

floscher replied:
Thanks for the advice. I'll look into it tomorrow.

aroben commented:
The changes look good but I have one small suggestion above.

arfon commented:
@floscher - would you mind rebasing against master too? It doesn't look like we can get a clean merge at this point.

floscher commented:
Done.
I've rebased against master and fixed the issue that json-files were passed through csonc.

arfon commented:
Looks good to me :+1:. Let's wait for a final :+1: from @aroben before merging.

aroben commented on script/download-grammars:
```diff
@@ -143,6 +143,8 @@ def load_grammar(path)
     raise "Failed to convert CSON grammar '#{path}': #{$?.to_s}" unless $?.success?
     JSON.parse(cson)
+  when '.json'
+    JSON.parse(File.open(path, 'rb').read)
```
This could be simplified to `JSON.parse(File.read(path))` I believe.

aroben commented:
This looks great! One more teensy suggestion above and then I think we're good.

floscher commented:
Thank you for the suggestion. I've incorporated the change into the pull request and rebased again on 8430f694e5.

Please note that the script currently doesn't run through. But this is another issue for which I opened #1958.

floscher commented:
You're welcome.
Is there a way to rerun Travis on the above commits? The build should pass now, as #1959 is merged.

aroben commented:
http://docs.travis-ci.com/user/pull-requests/ makes it sound like the only way is to update the PR by pushing again.

floscher commented:
Thanks for the hint! I think I'll simply rebase.

aroben commented:
Awesome! Thanks so much @floscher.

aroben commented:
This is now live on github.com.

floscher commented:
> This is now live on github.com.

Wow, that was quick!
Thank you @arfon and @aroben for the guidance concerning this and the related pull requests and for merging those so fast.
